datalad/distribution/uninstall.py

```diff
@@ -26,84 +27,151 @@
+                submodule = [sm for sm in ds.repo.repo.submodules
+                             if sm.path == subds_relpath][0]
+                if kill:
+                    lgr.debug("Unregister subdataset '%s' from %s",
+                              subds_relpath, ds)
+                    submodule.remove()
+                else:
+                    # put back empty dir as mount point
+                    os.makedirs(subds_path)
+                    ds.repo.add(subds_relpath, git=True)
+            else:
+                ds.repo.add(subds_relpath, git=True)
+        else:
+            out[subds_path] = ds2save[subds_path]
+    return out
 
 
 class Uninstall(Interface):
-    """Uninstall a dataset component or entire dataset(s)
+    """Uninstall dataset component(s) or entire dataset(s)
+
+    This command can be used to remove any installed dataset content. This
+    includes the content of files, as well as file handles, subdatasets, and
+    datasets themselves.
+
+    As datasets are relatively lightweight already, the default mode of
+    operation is to drop file content only, which could be re-installed later
+    on given an available remote source. The availability of at least one
+    remote copy is verified, by default, before file content is dropped. As
+    these checks could lead to slow operation (network latencies, etc), they
+    can be disabled.
+
+    Removing the handle of a subdataset will cause all corresponding file
+    content to be dropped, and all contained file handles to be removed.
+    However, the uninstalled subdataset remains registered in a potential
+    superdataset. Via the optional kill switch this association can be removed
+    as well, in which case the former subdataset is no longer part of the
+    superdataset, and cannot be reinstalled via a dataset handle again.
+
+    In contrast, uninstallation of file handles immediately yields their
+    complete removal from the corresponding dataset without the need of an
+    additional kill switch.
+
+    Any number of paths to process can be given as input. Recursion into
+    subdatasets needs to be explicitly enabled, while recursion in
+    subdirectories within a dataset is always done automatically. An optional
+    recursion limit is applied relative to each given input path.
+
+    The result of all handle removal operations is automatically saved in the
+    respective datasets.
 
+    Examples
+    --------
+
+    Drop all file content in a dataset::
+
+      ~/some/dataset$ datalad uninstall
+
+    Drop all file content in a dataset and all its subdatasets::
+
+      ~/some/dataset$ datalad uninstall --recursive
+
+    Uninstall a subdataset (undo installation)::
+
+      ~/some/dataset$ datalad uninstall --remove-handles somesubdataset1
+
+    Permanently remove a subdataset from a dataset and wipe out the subdataset
+    association too::
+
+      ~/some/dataset$ datalad uninstall --kill somesubdataset1
```

Comment:
I have pushed some minor doc fixups, but overall I was just left more confused...
1. `--kill` seems have transformed (according to doc) to gain a new  functionality of 'rm', i.e. removing the handle from git -- be it a subdataset or a file (handle)
But it actually doesn't seems to do it in case of sub-dataset
```
$> which datalad
/home/yoh/proj/datalad/datalad/venv-tests/bin/datalad
(venv-tests) 2 10447.....................................:Mon 14 Nov 2016 06:55:48 PM EST:.
(git)hopa:~/datalad/openfmri[master]git
$> datalad uninstall --kill  ds000105 
1 item was uninstalled:
Dataset: /home/yoh/datalad/openfmri/ds000105

changes on filesystem:                                                                                                           
 ds000105 | 1 -
```
so there were no changes to .gitmodules
2. I just feel that we are in the same (although easier) situation as with former "do it all" install.  As you can see from your own description above it does perform "Drop" operation.  And `--kill` seems to perform `rm`.

So I wondered why don't we really have `rm`, `drop`, and `uninstall`, where it a clear analog of `add`, `get`, and `install` commands? ... still need to think about it though...